documentation/PBC_APP_APDU.md

```diff
@@ -0,0 +1,177 @@
+# Technical Specification of Partisia Blockchain Ledger App
+
+## About
+
+This documentation describes the APDU messages interface to communicate with Partisia Blockchain Ledger App.
+
+The application covers the following functionalities :
+
+  - Get a blockchain address given a [BIP 32 path](https://bips.dev/32/)
+  - Sign a basic PBC transaction given a BIP 32 path and raw transaction
+  - Retrieve the PBC app version
+  - Retrieve the PBC app name
+
+The application interface can be accessed over HID or BLE.
+
+## APDUs
+
+### GET PBC PUBLIC ADDRESS
+
+#### Description
+
+This command returns the blockchain address for the given BIP 32 path.
+
+The address can be optionally checked on the device before being returned.
+
+#### Coding
+
+##### `Command`
+
+| CLA | INS   | P1                                                 | P2    | Lc       | Le       |
+| --- | ---   | ---                                                | ---   | ---      | ---      |
+| `E0` | `05`  | `00`: return address                               |`00`   | variable | variable |
+|     |       | `01`: display address and confirm before returning |       |          |          |
+
+##### `Input data`
+
+| Description                                                      | Length |
+| ---                                                              | ---    |
+| Number of BIP 32 derivations to perform (max 10)                 | 1      |
+| First derivation index (big endian)                              | 4      |
+| ...                                                              | 4      |
+| Last derivation index (big endian)                               | 4      |
+
+##### `Output data`
+
+| Description                                                      | Length |
+| ---                                                              | ---    |
+| Blockchain address                                               | 21     |
+
+
+### SIGN PBC TRANSACTION
+
+#### Description
+
+This command signs a PBC transaction after having the user validate the transactions parameters.
+
+The input data is the transaction streamed to the device in 255 bytes maximum data chunks. The ID of the chain to sign for must be sent in the first block.
+
+#### Coding
+
+##### `Command`
+
+| CLA | INS  | P1                   | P2                               | Lc       | Le       |
+| --- | ---  | ---                  | ---                              | ---      | ---      |
+|`E0` |`06`  | `00`-`FF` : chunk index | `00` : last transaction data block | variable | variable |
+|     |      |                      | `80` : subsequent transaction data block |    |          |
```

Comment:
'80' is the be sent for anything except for the last block, 'subsequent' seems like the wrong terminology.